modules/ca/src/client/comBuf.h

```diff
@@ -208,6 +209,11 @@ inline unsigned comBuf :: push ( const epicsUInt8 *pValue, unsigned nElem )
     return copyInBytes ( pValue, nElem );
 }
 
+inline unsigned comBuf :: push ( const char *pValue, unsigned nElem )
+{
+    return copyInBytes ( pValue, nElem );
+}
+
```

Comment:
I don't like adding a third specialization here.  A simpler solution would be to combine the `epicsInt8` and `epicsUInt8` specializations into the `T` definition with a condition on `sizeof(T)==1`.

```c++
template < class T >
inline bool comBuf :: push ( const T & value )
{
    if(sizeof(T)==1)
        return copyInBytes ( pValue, nElem );
```

If adding a third specialization is somehow necessary, then the argument types should be the three plain C types.

Reply:
Maybe. I tried what works. As the function is `inline`, the run-time overhead is zero.
Not sure if your solution adds run-time overhead. And branching on types does not seem like good C++ style to me.

Reply:
The "ideal" solution would be one partial specialization which covers the three `sizeof(T)==1` types.  cf. [SFINAE](https://en.cppreference.com/w/cpp/language/sfinae) and [`std::enable_if`](https://en.cppreference.com/w/cpp/types/enable_if).  I'm not going to insist that you figure out this fairly tricky pattern just for this change.  Unless optimization is completely disabled (`-O0`), branching on a constant has no runtime cost.  I've verified this on various GCC versions.